devices/xiaomi-beryllium/firmware/default.nix

```diff
@@ -0,0 +1,19 @@
+{ lib
+, fetchFromGitLab
+, runCommand
+}:
+
+let
+  baseFw = fetchFromGitLab {
+    owner = "sdm845-mainline";
+    repo = "firmware-xiaomi-beryllium";
+    rev = "66ec241f43f26e8e8a2c6f18858c92f1a7fb16e4";
+    sha256 = "sha256-7CaXWOpao+vuFA7xknzbLml2hxTlmuzFCEM99aLD2uk=";
+  };
+in runCommand "xiaomi-sdm845-firmware" { inherit baseFw; } ''
+  mkdir -p $out/lib/firmware
+  cp -r $baseFw/lib/firmware/* $out/lib/firmware/
+  chmod +w -R $out
+  # rm -rf $out/lib/firmware/postmarketos
+  # cp -r $baseFw/lib/firmware/postmarketos/* $out/lib/firmware
```

Comment:
Copied this over from oneplus-enchilada. However the structure is quite different so I'm not even (close) to being sure this does anything meaningfull or that these firmware files are still on my device after flashing.